### PR TITLE
feat(sdk): enabling tf-gcp test command

### PR DESCRIPTION
**Pull request by tsuf239**
## Description
closes: https://github.com/winglang/wing/issues/4403
- Adding inflight function, and enabling test command to tf-GCP
- Refactoring tf-gcp permissions (now each function has its service account, and the permissions granted to it)
- chore: minor code formatting in the HTTP module

image
## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/shared-gcp/function.inflight.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * Invoke the function, passing the given payload as an argument.
+   *  @returns the function returned payload only
+   */
+  public async invoke(payload: string): Promise<string> {
+    const url = `https://${this.region}-${this.projectId}.cloudfunctions.net/${this.functionName}`;
+    try {
+      const auth = new GoogleAuth();
+      const client = await auth.getIdTokenClient(url);
+
```
Comment (hasanaburayyan) [inline libs/wingsdk/src/shared-gcp/function.inflight.ts:49]: Im curious, not sure I see it in the code (maybe its a GCP thing) but what permission does the function require in order to make this `auth.getIdTokenClient(url)`?

**Review by hasanaburayyan — APPROVED** [review]
LGTM I added a few nits and questions

File: libs/wingsdk/src/shared-gcp/function.inflight.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+   * Invoke the function, passing the given payload as an argument.
+   *
+   * @returns the function returned payload and logs
+   */
+  public async invokeWithLogs(
+    payload: string,
+    token: string
+  ): Promise<[string, Trace[]]> {
+    console.error("Test invocation on tf-gcp doesn't include logs yet");
+    //TODO: add traces to tf-azure tests- https://github.com/winglang/wing/issues/4904
```
Comment (staycoolcall911) [inline libs/wingsdk/src/shared-gcp/function.inflight.ts:87]:
```suggestion
    //TODO: add traces to tf-gcp tests- https://github.com/winglang/wing/issues/4904
```

**Comment by tsuf239** [comment]
tested again, the test command is working both for failing and successful tests ✨ 
image
image

**Pull request merged**